internal/csi-addons/rbd/replication.go

```diff
@@ -637,14 +663,53 @@ func (rs *ReplicationServer) ResyncVolume(ctx context.Context,
 		ready = checkRemoteSiteStatus(ctx, mirrorStatus)
 	}
 
-	err = rbdVol.ResyncVol(localStatus, req.Force)
+	creationTime, err := rbdVol.GetImageCreationTime()
 	if err != nil {
-		return nil, getGRPCError(err)
+		return nil, status.Errorf(codes.Internal, "failed to get image info: %s", err.Error())
 	}
 
-	err = checkVolumeResyncStatus(localStatus)
+	// image creation time is stored in the image metadata. it looks like
+	// `"seconds:1692879841 nanos:631526669"`
+	savedImageTime, err := rbdVol.GetMetadata(imageCreationTimeKey)
 	if err != nil {
-		return nil, status.Error(codes.Internal, err.Error())
+		return nil, status.Errorf(codes.Internal, "failed to get image creation time: %s", err.Error())
+	}
+
+	parts := strings.Fields(savedImageTime)
+	if len(parts) != 2 {
+		return nil, status.Errorf(codes.Internal, "failed to parse image creation time: %s", savedImageTime)
+	}
+	if len(strings.Split(parts[0], ":")) != 2 || len(strings.Split(parts[1], ":")) != 2 {
+		return nil, status.Errorf(codes.Internal, "failed to parse image creation time: %s", savedImageTime)
+	}
+	secondsStr := strings.Split(parts[0], ":")[1]
```

Comment:
In my opinion it would be to good to have a different function to parse the creation time and get the required format, we can also further add unit test for it. What do you think @Madhu-1 ?

Reply:
yes, definitely a good suggestion!